docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,30 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when the user wants to use his own
+tool-specific config, to increase performance. If a config file is supplied
+by user that means he won't need the setting provided by coala to generate
+the config file. Therefore, in such scenarios, it is advised to return
+``None`` in ``generate_config()``.
```

Comment:
Hm this still reads a bit complicated, I'm still not convinced... I'll wrap up my head again for this, please fix the other issues already ;)